rosbag2_cpp/test/rosbag2_cpp/test_multifile_reader.cpp

```diff
@@ -114,3 +127,95 @@ TEST_F(MultifileReaderTest, get_all_topics_and_types_throws_if_no_storage)
 {
   EXPECT_ANY_THROW(reader_->get_all_topics_and_types());
 }
+
+class FakeSequentialReader : public rosbag2_cpp::readers::SequentialReader
+{
+public:
+  FakeSequentialReader(
+    std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory,
+    std::shared_ptr<rosbag2_cpp::SerializationFormatConverterFactoryInterface> converter_factory,
+    std::unique_ptr<rosbag2_storage::MetadataIo> metadata_io)
+  : SequentialReader(std::move(storage_factory),
+      std::move(converter_factory),
+      std::move(metadata_io)) {}
+
+  void decompress_message(rosbag2_storage::SerializedBagMessage *) override
+  {
+    decompress_message_called = true;
+  }
+
+  void decompress_file(const std::string &) override
+  {
+    decompress_file_called = true;
+  }
+
+  bool decompress_message_called = false;
+  bool decompress_file_called = false;
+};
+
+class ReaderCompressionTest : public Test
```

Comment:
There's a lot of local variables that could be const in this class and the following tests.